src/test/scala/chiselTests/experimental/hierarchy/SeparateElaborationSpec.scala

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chiselTests.experimental.hierarchy
+
+import chiselTests.ChiselFunSpec
+import chisel3._
+import chisel3.stage.{ChiselGeneratorAnnotation, ChiselStage, DesignAnnotation}
+import chisel3.internal.ImportedDefinitionAnnotation
+import chisel3.experimental.hierarchy.{Definition, Instance}
+import firrtl.options.TargetDirAnnotation
+
+import java.nio.file.Paths
+import scala.io.Source
+
+class SeparateElaborationSpec extends ChiselFunSpec with Utils {
+  import Examples._
+
+  describe("(0): Elaborating an Instance separately from its Definition") {
+    it("should result in an instantiation in FIRRTL without a module declaration.") {
+      val testDir = createTestDirectory(this.getClass.getSimpleName).toString
+      val dutAnnos = (new ChiselStage).run(
+        Seq(
+          ChiselGeneratorAnnotation(() => new AddOne),
+          TargetDirAnnotation(testDir)
+        )
+      )
+
+      // Grab DUT definition to pass into testbench
+      val designDefs = dutAnnos.flatMap { a =>
+        a match {
+          case a: DesignAnnotation[_] =>
+            Some(a.design.asInstanceOf[AddOne].toDefinition)
+          case _ => None
+        }
+      }
+      require(designDefs.length == 1, s"Exactly one DesignAnnotation should exist, but found: $designDefs.")
+      val dutDef = designDefs.head
+
+      class Testbench(defn: Definition[AddOne]) extends Module {
+        // Make sure names do not conflict
+        val mod = Module(new AddOne)
+        val inst = Instance(defn)
+
+        // Tie inputs to a value so ChiselStage does not complain
+        mod.in := 0.U
+        inst.in := 0.U
+      }
+
+      (new ChiselStage).run(
+        Seq(
+          ChiselGeneratorAnnotation(() => new Testbench(dutDef)),
+          TargetDirAnnotation(testDir),
+          ImportedDefinitionAnnotation(dutDef)
+        )
+      )
+
+      // Check that the output RTL has only a module instantiation and no
+      // module declaration.
+      val tb_rtl = Source.fromFile(s"$testDir/Testbench.v").getLines.mkString
+      tb_rtl should include("AddOne inst (")
+      (tb_rtl should not).include("module AddOne")
+    }
+  }
+
+  describe("(1): Elaborating an Instance and Definition together") {
+    it("should not result in a repeat definition of the module.") {
+      val testDir = createTestDirectory(this.getClass.getSimpleName).toString
+
+      class Top extends Module {
+        val inst = Instance(Definition(new AddOne))
```

Comment:
This is not testing what we talked about. Here, you are creating a definition, then instantiating it. This should always work (regardless of your changes in your PR).

We care about the following case:

```
class Top(dut: Definition[DUT]) {
  val i0 = Instance(Definition(new DUT))
  val i1 = Instance(dut)
}
```

We need to make sure that i0's definition's module's name is different than i1's definition's module's name. Given your code, i1's module name will get the original dut's definition's module's name. We need the test to actually test this.

Reply:
I wrote this test to make sure that if we're doing things the "regular" way, an `ExtModule` is NOT created by `Instance`... But now that you point out this is the regular case, this test case does seem redundant with our regular D/I tests; I will remove it and add the case you stated!